code/src/context/ModalContext.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+  ReactElement,
+  ReactNode,
+  createContext,
+  useContext,
+  useRef,
+  useState
+} from "react";
+import Dialog from "@mui/material/Dialog";
+import AppBox from "@/components/app-box/AppBox";
+
+type ModalContextType = {
+  handleOpenModal: (component: ReactElement) => void;
+  handleCloseModal: () => void;
+};
+
+type ModalProviderProps = { children: ReactNode };
+
+const ModalContext = createContext<ModalContextType | null>(null);
+
+const ModalProvider = ({ children }: ModalProviderProps) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const modalRef = useRef<ReactElement | null>(null);
+  const handleOpenModal = (component: ReactElement) => {
+    modalRef.current = component;
+    setIsModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+    modalRef.current = null;
+  };
+
+  return (
+    <ModalContext.Provider value={{ handleOpenModal, handleCloseModal }}>
+      {modalRef.current && (
+        <Dialog open={isModalOpen} onClose={handleCloseModal}>
+          <AppBox>{modalRef.current}</AppBox>
+        </Dialog>
+      )}
+      {children}
```

Comment:
too long logic here
create variable for `modalRef` here. 
But for now it's not readable

Reply:
removed ref